src/service/extensions-impl.js

```diff
@@ -239,19 +239,22 @@ export class Extensions {
   /**
    * Reloads the new version of the extension.
    * @param {string} extensionId
-   * @return {!Promise<!ExtensionDef>}
    */
   reloadExtension(extensionId) {
     // Ignore inserted script elements to prevent recursion.
     const els = this.getExtensionScripts_(
       extensionId,
       /* includeInserted */ false
     );
-    devAssert(
-      els.length > 0,
-      'Cannot find script for extension: %s',
-      extensionId
-    );
+    if (!els.length) {
+      const TAG = 'reloadExtension';
+      user().warn(
```

Comment:
There's already a different error for this: `Error: amp-fit-text is already registered. The script tag for amp-fit-text is likely included twice in the page.`

Happens in custom-element-registry.js when we try to register a CE twice.

Reply:
That happens too late, though, since any side-effects in the code can already execute. I think we need an earlier check in `runtime.js` when we're processing the pushed `ExtensionPayload`.

Reply:
Right and it doesn't cover non-CE extensions. I guess we have a similar check for templates. #29050

Reply:
This got off topic from the request. Let's make this an error, not a warn.

We can solve the escape hatch in a follow up.